PF/PF-Assignment3/JobScheduler.java

```diff
@@ -0,0 +1,134 @@
+import java.util.*;
+import java.util.List;
+
+public class JobScheduler {
+	
+	private int waiting_Time;
+	private int completion_Time;
+	private int turn_Around_Time;
+	private int last_Completed_Time;
+	private int arrival_Time;
+	private int burst_Time;
+	
+	// Initialising the value of arrival time and burst time for each process
+	public JobScheduler(int arrival_Time,int burst_Time)
+	{
+		this.arrival_Time = arrival_Time;
+		this.burst_Time = burst_Time;
+	}
+	
+	// Calculating completion time for the process
+	public int completionTime(JobScheduler jobs)
+	{
+		jobs.completion_Time = jobs.arrival_Time + jobs.waiting_Time + jobs.burst_Time;
+		
+		return jobs.completion_Time;
+	}
+	
+	// Calculating waiting time for the process
+	public int waitTime(JobScheduler jobs,JobScheduler previousJob)
+	{
+		if(jobs.arrival_Time < previousJob.last_Completed_Time)
+		{
+			jobs.waiting_Time = previousJob.last_Completed_Time - jobs.arrival_Time;
+			jobs.last_Completed_Time = previousJob.last_Completed_Time + jobs.burst_Time;
+		}
+		else
+		{
+			jobs.waiting_Time = 0;
+			jobs.last_Completed_Time = previousJob.last_Completed_Time + (jobs.arrival_Time - previousJob.last_Completed_Time) + jobs.burst_Time;
+		}
+		return jobs.waiting_Time;
+	}
+	
+	//Calculating turn around time for the process
+	public int turnAroundTime(JobScheduler jobs)
+	{
+		jobs.turn_Around_Time = jobs.completion_Time - jobs.arrival_Time;
+		return jobs.turn_Around_Time;
+	}
+	
+	// Calculating average waiting time for the process
+	public static float avgWaitingTime(List<JobScheduler> job,int pro)
```

Comment:
This method can be implemented in a separate Utils class as it is processing array of this object.